src/hooks/useUserProfile.tsx

```diff
@@ -1,11 +1,38 @@
-import React, { createContext, PropsWithChildren, useEffect, useState } from 'react';
+import React, { createContext, PropsWithChildren, useCallback, useEffect, useState } from 'react';
 import { createUserProfile, getUserProfile } from '../api';
-import { UserProfile } from '../types';
+import { BrightIdVerificationStatus, UserProfile } from '../types';
 
-export const UserProfileContext = createContext<UserProfile | null>(null);
+export const UserProfileContext = createContext<{
+  userProfile: UserProfile | null;
+  refreshUserProfile: (()=> void) | null;
+  loading: boolean;
+}>({userProfile:null,refreshUserProfile:null,loading: false});
 
 export function UserProfileProvider({ children, address }: PropsWithChildren<{ address: string | null | undefined }>) {
   const [userProfile, setUserProfile] = useState<UserProfile | null>(null);
+
+  const [loading, setLoading] = useState(false);
+
+  const refreshUserProfile = useCallback(async () => {
+    if (userProfile?.verificationStatus === BrightIdVerificationStatus.VERIFIED || loading) {
+      return;
+    }
```

Comment:
This is not part of "refreshProfile" functionality. should be handled in the component itself